### PR TITLE
Use a new approach to aligning labels with their table rows.  #1532

### PR DESCRIPTION
**Pull request by dpvc**
Use a new approach to aligning labels with their table rows that should reduce the round-off problems by forcing each row to the desired height, and doing the same with the labels.  Resolves issue #1532 (and #1500).

### CODE REVIEW

File: unpacked/jax/output/CommonHTML/autoload/mtable.js [hunk 1]
```diff
@@ -485,21 +475,18 @@ MathJax.Hub.Register.StartupHook("CommonHTML Jax Ready",function () {
...
       for (var i = 0, m = LABELS.length; i < m; i++) {
-        if (!state.HD) {h = H[i]; d = D[i]}
         if (LABELS[i] && this.data[i].data[0]) {
           labels.appendChild(LABELS[i]);
           var lbox = this.data[i].data[0].CHTML;
-          T += h - Math.max(1,lbox.h);
-          if (T) LABELS[i].style.marginTop = CHTML.Em(T);
-          T = d - lbox.d;
+          T = state.RH[i] - Math.max(1,lbox.h);
+          if (T) LABELS[i].firstChild.firstChild.style.marginTop = CHTML.Em(T);
```
Comment (zorkow) [inline unpacked/jax/output/CommonHTML/autoload/mtable.js]: These are all guaranteed to exist?

File: unpacked/jax/output/CommonHTML/autoload/mtable.js [hunk 2]
```diff
@@ -110,6 +105,12 @@ MathJax.Hub.Register.StartupHook("CommonHTML Jax Ready",function () {
           if (cbox.w > W[j]) W[j] = cbox.w;
         }
       }
+      if (values.equalrows) {
+        state.HD = true;
+        var HH = Math.max.apply(Math,H);
+        var DD = Math.max.apply(Math,D);
+        for (var i = 0, m = H.length; i < m; i++) {H[i] = HH; D[i] = DD}
```
Comment (zorkow) [inline unpacked/jax/output/CommonHTML/autoload/mtable.js]: I assume D and H are of the same length?

File: unpacked/jax/output/CommonHTML/autoload/mtable.js (hunk 1 above)
Comment (dpvc) [inline unpacked/jax/output/CommonHTML/autoload/mtable.js]: If you mean the `first child.firstchild` then yes.  if you mean the various array elements indexed by `[i]`, then yes.

File: unpacked/jax/output/CommonHTML/autoload/mtable.js (hunk 2 above)
Comment (dpvc) [inline unpacked/jax/output/CommonHTML/autoload/mtable.js]: Yes.  `H[i]` and `D[i]` are the heights and depths of row `i`, so they have the same number of elements.

**Comment by zorkow** [comment]
LGTM.

**Pull request merged**